packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -0,0 +1,744 @@
+	}
+
+	public get type(): TreeNodeSchemaIdentifier {
+		return this.schema.name;
+	}
+
+	/**
+	 * Set this node's parentage (see {@link FlexTreeNode.parentField}).
+	 * @remarks A node may only be adopted to a new parent one time, though this method may be called idempotently with the same parent and index.
+	 */
+	public adopt(parent: MapTreeField<FlexAllowedTypes>, index: number): void {
+		assert(
+			this.location === undefined ||
+				(parent.isSameAs(this.location.parent) && index === this.location.index),
+			"Node may not be reparented",
+		);
+		this.location = { parent, index };
+	}
+
+	/**
+	 * The field this tree is in, and the index within that field.
+	 * @remarks If this node is unparented, this method will return the special {@link rootMapTreeField} as the parent.
+	 */
+	public get parentField(): LocationInField {
+		if (this.location === undefined) {
+			return {
+				parent: rootMapTreeField,
+				index: -1,
+			};
+		}
+
+		return this.location;
+	}
+
+	public is<TSchemaInner extends FlexTreeNodeSchema>(
+		schema: TSchemaInner,
+	): this is FlexTreeTypedNode<TSchemaInner> {
+		return (schema as unknown) === this.schema;
+	}
+
+	public tryGetField(key: FieldKey): MapTreeField<FlexAllowedTypes> | undefined {
+		const field = this.mapTree.fields.get(key);
+		if (field !== undefined && field.length > 0) {
+			return getOrCreateField(this, key, field, this.schema.getFieldSchema(key));
+		}
+	}
+
+	public getBoxed(key: string): FlexTreeField {
+		const fieldKey: FieldKey = brand(key);
+		const field = this.mapTree.fields.get(fieldKey) ?? [];
+		return getOrCreateField(this, fieldKey, field, this.schema.getFieldSchema(fieldKey));
+	}
+
+	public boxedIterator(): IterableIterator<FlexTreeField> {
+		return mapIterable(this.mapTree.fields.entries(), ([key, field]) =>
+			getOrCreateField(this, key, field, this.schema.getFieldSchema(key)),
+		);
+	}
+
+	public treeStatus(): TreeStatus {
+		return TreeStatus.Created;
+	}
+
+	public get value(): Value {
+		return this.mapTree.value;
+	}
+
+	public on<K extends keyof FlexTreeNodeEvents>(
+		eventName: K,
+		listener: FlexTreeNodeEvents[K],
+	): () => void {
+		return this.events.on(eventName, listener);
+	}
+
+	public get context(): FlexTreeContext {
+		throw unsupportedError("Getting context");
+	}
+
+	public get anchorNode(): AnchorNode {
+		throw unsupportedError("Reading anchor node");
+	}
+
+	private walkTree(): void {
+		for (const [key, mapTrees] of this.mapTree.fields) {
+			const field = getOrCreateField(this, key, mapTrees, this.schema.getFieldSchema(key));
+			for (let index = 0; index < field.length; index++) {
+				const child = getOrCreateChild(
+					mapTrees[index],
+					this.schema.getFieldSchema(key).allowedTypes,
+					{ parent: field, index },
+				);
+				// These next asserts detect the case where `getOrCreateChild` gets a cache hit of a different node than the one we're trying to create
+				assert(child.location !== undefined, "Expected child to have parent");
```

Comment:
Nit (the other asserts call this "node" instead of "child")
```suggestion
				assert(child.location !== undefined, "Expected node to have parent");
```

Reply:
Nice, thanks.